File: ios/BankShared/Sources/BankShared/Models/Operation.swift
import Foundation

public struct Operation: Codable, Identifiable, Sendable {
    public let id: Int64
    public let accountId: Int64
    public let type: OperationType
    public let amount: Decimal
    public let currency: Currency
    public let relatedAccountId: Int64?
    public let exchangeRate: Decimal?
    public let description: String?
    public let createdAt: String
}

File: frontend/src/entities/operation.ts
import type { Currency, OperationType } from './common'

export interface OperationResponse {
  id: number
  accountId: number
  type: OperationType
  amount: number
  currency: Currency
  relatedAccountId: number | null
  exchangeRate: number | null
  description: string | null
  createdAt: string
}
